src/data/set/intervals/basic.lean

```diff
@@ -296,6 +296,17 @@ lemma Ici_inter_Iio : Ici a ∩ Iio b = Ico a b := rfl
 lemma Ioi_inter_Iic : Ioi a ∩ Iic b = Ioc a b := rfl
 lemma Ioi_inter_Iio : Ioi a ∩ Iio b = Ioo a b := rfl
 
+/-- In a dense order `α`, open intervals `Ioo a b` are also densely ordered. -/
+instance Ioo_densely_ordered [densely_ordered α] {a b : α} :
+  densely_ordered (set.Ioo a b) :=
```

Comment:
Do we have a name for the following property? 
```lean
def is_interval (s : set α) : Prop := ∀ (x y ∈ s) z, x < z → z < y → z ∈ s
```
I'm not sure about the name `is_connected` might be another potential name, but that sounds very topological. It is at least the property shared by all intervals.
Under that condition you should be able to prove that `s` is also densely ordered, which would reduce proof duplication, since all intervals have this property.